File: frontend/src/services/leaderboardService.ts
import api from './api';

const leaderboardService = {
  getStudentLeaderboard: () => api.get('/leaderboard/students'),
  getDepartmentLeaderboard: () => api.get('/leaderboard/departments'),
  getFacultyLeaderboard: () => api.get('/leaderboard/faculty'),
  getSubjectLeaderboard: () => api.get('/leaderboard/subjects'),
};

export default leaderboardService;

File: backend/src/main/java/com/evalorithm/service/LeaderboardService.java
package com.evalorithm.service;

import com.evalorithm.dto.response.LeaderboardResponse;

import java.util.List;
import java.util.Map;

public interface LeaderboardService {

    List<Map<String, Object>> getDepartmentLeaderboard();

    List<LeaderboardResponse> getStudentLeaderboard(int limit);

    List<Map<String, Object>> getFacultyLeaderboard();

    List<Map<String, Object>> getSubjectLeaderboard();
}
